docs/proposals/107-spark-client/README.md

```diff
@@ -0,0 +1,1023 @@
+
+    # Interactive analysis
+    by_product = sales_df.groupBy("product").sum("revenue")
+    top_products = by_product.orderBy("sum(revenue)", ascending=False).limit(10)
+
+    # View results
+    for row in top_products.collect():
+        print(f"{row.product}: ${row['sum(revenue)']:,.2f}")
+
+    # Export for pipeline
+    session.export_to_pipeline_artifact(
+        top_products,
+        "/outputs/top_products.parquet"
+    )
+
+    # Check metrics
+    metrics = session.get_metrics()
+    print(f"Total queries: {metrics.queries_executed}")
+```
+
+### Example: Notebook Workflow
+
+```python
+# Cell 1: Setup
+from kubeflow.spark import SparkClient, ConnectBackendConfig
+
+config = ConnectBackendConfig(connect_url="sc://localhost:15002")
+client = SparkClient(backend_config=config)
+session = client.create_session("notebook-analysis")
+
+# Cell 2: Load and explore
+df = session.sql("SELECT * FROM customers LIMIT 100")
+df.show()
+
+# Cell 3: Feature engineering
+features = df.select("customer_id", "age", "spend_total")
+features = features.withColumn("spend_per_year", features.spend_total / features.age)
+
+# Cell 4: Analysis
+summary = features.describe()
+summary.show()
+
+# Cell 5: Cleanup
+session.close()
+```
+
+### Integration with Other Backends
+
+Combined workflow example:
+
+```python
+from kubeflow.spark import SparkClient, ConnectBackendConfig, OperatorBackendConfig
+
+# Step 1: Interactive development with ConnectBackend
+connect_config = ConnectBackendConfig(connect_url="sc://dev-cluster:15002")
+dev_client = SparkClient(backend_config=connect_config)
+
+with dev_client.create_session("dev") as session:
+    # Test and validate query
+    test_df = session.sql("SELECT * FROM data LIMIT 1000")
+    test_df.show()
+    # Iterate and refine...
+
+# Step 2: Production batch job with OperatorBackend
+prod_config = OperatorBackendConfig(namespace="production")
+prod_client = SparkClient(backend_config=prod_config)
+
+# Submit validated job at scale
+response = prod_client.submit_application(
+    app_name="production-etl",
+    main_application_file="s3a://jobs/etl_pipeline.py",
+    driver_cores=4,
+    driver_memory="16g",
+    executor_cores=4,
+    executor_memory="32g",
+    num_executors=50,
+)
+```
+#### 4. Data Models
+
+**SparkApplicationResponse**:
+```python
+@dataclass
+class SparkApplicationResponse:
+    submission_id: str      # Unique submission ID
+    status: str             # Initial status
+    message: Optional[str]  # Optional message
+    timestamp: str          # Submission timestamp
+```
+
+**ApplicationStatus**:
+```python
+@dataclass
+class ApplicationStatus:
+    submission_id: str
+    state: ApplicationState  # Enum: NEW, RUNNING, COMPLETED, FAILED, etc.
+    app_id: Optional[str]    # Spark application ID
+    driver_info: Optional[Dict]
+    executor_info: Optional[Dict]
+    submission_time: Optional[str]
+    completion_time: Optional[str]
+    message: Optional[str]
+```
+
+### User Stories
```

Comment:
Could you move these user stories to the beginning of the KEP?
Here is the example: https://github.com/kubeflow/sdk/tree/main/docs/proposals/46-hyperparameter-optimization#user-stories

Reply:
Done https://github.com/kubeflow/sdk/pull/163/commits/90ba54e2b2f4fac1bd3817e41e66de14b54f96c7